fiks-meldingsformidler/src/main/java/no/difi/meldingsutveksling/ks/mapping/FiksMapper.java

```diff
@@ -1,26 +1,250 @@
+                .filter(Dokumentbeskrivelse.class::isInstance)
+                .map(Dokumentbeskrivelse.class::cast)
+                .collect(Collectors.toSet());
+    }
+
+    private Adresse getMottaker(NextMoveOutMessage message) {
+        final InfoRecord receiverInfo = serviceRegistry.getInfoRecord(message.getReceiverIdentifier());
+        return mottakerFrom(receiverInfo);
+    }
+
+    private Arkivmelding getArkivmelding(NextMoveOutMessage message) throws NextMoveException {
+        String arkivmeldingIdentifier = getArkivmeldingIdentifier(message);
+
+        try (FileEntryStream fileEntryStream = cryptoMessagePersister.readStream(message.getMessageId(), arkivmeldingIdentifier)) {
+            return ArkivmeldingUtil.unmarshalArkivmelding(fileEntryStream.getInputStream());
+        } catch (JAXBException e) {
+            throw new NextMoveException("Error unmarshalling arkivmelding", e);
+        } catch (IOException e) {
+            throw new NextMoveException("Reading failed for arkivmelding", e);
+        }
+    }
+
+    private String getArkivmeldingIdentifier(NextMoveOutMessage message) throws NextMoveException {
+        return message.getFiles().stream()
+                .filter(f -> ARKIVMELDING_FILE.equals(f.getFilename()))
+                .findAny()
+                .map(BusinessMessageFile::getIdentifier)
+                .orElseThrow(() -> new NextMoveException(format("No attachement \"%s\" found", ARKIVMELDING_FILE)));
+    }
+
+    private Set<Dokument> mapArkivmeldingDokumenter(NextMoveOutMessage message, Set<Dokumentbeskrivelse> docs, X509Certificate cert) {
+        return docs.stream()
+                .flatMap(p -> p.getDokumentobjekt().stream())
+                .map(d -> getBusinessMessageFile(message, d.getReferanseDokumentfil()))
+                .map(file -> getDocument(message.getMessageId(), file, cert))
+                .collect(Collectors.toSet());
+    }
+
+    private BusinessMessageFile getBusinessMessageFile(NextMoveOutMessage message, String referanseDokumentfil) {
+        return message.getFiles().stream()
+                .filter(bmf -> bmf.getFilename().equals(referanseDokumentfil))
+                .findFirst()
+                .orElseThrow(() -> new NextMoveRuntimeException(
+                        String.format("File '%s' referenced in '%s' not found", referanseDokumentfil, message.getMessageId())));
+    }
+
+    private Dokument getDocument(String messageId, BusinessMessageFile file, X509Certificate cert) {
+        try {
+            FileEntryStream fileEntryStream = cryptoMessagePersister.readStream(messageId, file.getIdentifier());
+
+            return Dokument.builder()
+                    .withData(getDataHandler(cert, fileEntryStream.getInputStream()))
+                    .withFilnavn(file.getFilename())
+                    .withMimetype(file.getMimetype())
+                    .build();
+        } catch (IOException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)